app/Http/Requests/UpdateAssetRequest.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace App\Http\Requests;
+
+use App\Models\Asset;
+use Illuminate\Support\Facades\Gate;
+use Illuminate\Validation\Rule;
+
+class UpdateAssetRequest extends ImageUploadRequest
+{
+    /**
+     * Determine if the user is authorized to make this request.
+     *
+     * @return bool
+     */
+    public function authorize()
+    {
+        return Gate::allows('update', new Asset);
+    }
+
+    /**
+     * Get the validation rules that apply to the request.
+     *
+     * @return array
+     */
+    public function rules()
+    {
+        $rules = array_merge([
+            'model_id' => ['integer', 'exists:models,id,deleted_at,NULL', 'not_array'],
```

Comment:
I get these three rules - where we are removing the 'required' attribute. But a lot of the rest of the rules look exactly the same as the rules on the model. Can we do a kind-of 3-way `array_merge()` here, where we grab the Asset's `$rules` (or maybe `rules()`?), override these couple of ones where we need to pull the `required` attribute, then grab the `parent::rules()` and merge those in too? I feel like that would cut down on the duplication. Or, another way you could go - grab the `Asset::rules`, and grep-out the 'required' attributes, and merge in the parent::rules()?

I just hate repeating rules in multiple places and fear that if we change some constraint somewhere - let's say, we decide to make 'name' have a max of 1023 characters for example - then having to go and update that in 4 or more places. That feels like a foot-gun waiting to happen.

Reply:
Alright, I've put a couple of different options for resolving this in the `rules()` method - `$rules` & `$rules2`

One quick thing, the conversion of the rule strings to arrays in the model is not necessary if we go with option 1 (`$rules`) - converting them to arrays just makes it a little easier to manipulate.

Reply:
I like `$rules` because I find it much easier to read, but if you were really stuck on option 2, I would be OK with it.